src/components/layout/footer.tsx

```diff
@@ -0,0 +1,113 @@
+"use client";
+
+import copyrightLogo from "@public/images/icons/copyright.svg";
+import githubIcon from "@public/images/icons/github.svg";
+import linkedinIcon from "@public/images/icons/linkedin.svg";
+import slackIcon from "@public/images/icons/slack.svg";
+import Image from "next/image";
+import React, { CSSProperties, useEffect, useState } from "react";
+
+const Footer = React.memo(() => {
+  const [isMobile, setIsMobile] = useState(false);
+
+  useEffect(() => {
+    const handleResize = () => {
+      setIsMobile(window.innerWidth <= 768);
+    };
+
+    window.addEventListener("resize", handleResize);
+    handleResize();
+
+    return () => window.removeEventListener("resize", handleResize);
+  }, []);
```

Comment:
It would be good idea to move the logic out of the component ! Some new hook.

Reply:
I have updated footer to leverage tailwind classes so this is unnecessary now